src/components/marquee-wizard/marquee-wizard.component.spec.ts

```diff
@@ -276,95 +273,146 @@ describe('Marquee wizard with delayed step creation', () => {
+                header="Step Three"
+                [valid]="step3Valid"
+                [(visited)]="step3Visited"
+                [(completed)]="step3Completed"
+                (visitedChange)="visitedChanged(2, $event)">
+                Step Three Content
+            </ux-marquee-wizard-step>
         </ux-marquee-wizard>
     `
 })
 export class MarqueeWizardValidationComponent {
     currentStep: number;
     resetVisitedOnValidationError = true;
+
+    // step 1 values
     step1Valid = true;
     step1Visited: boolean;
     step1Completed: boolean;
+
+    // step 2 values
+    step2Valid: boolean = true;
     step2Visited: boolean;
     step2Completed: boolean;
+
+    // step 3 values
+    step3Valid: boolean = true;
+    step3Visited: boolean;
+    step3Completed: boolean;
+
+    visitedChanged(index: number, value: boolean): void {
+    }
 }
 
 describe('Marquee wizard with validation', () => {
     let component: MarqueeWizardValidationComponent;
     let fixture: ComponentFixture<MarqueeWizardValidationComponent>;
     let nativeElement: HTMLElement;
+    let visitedChanged: jasmine.Spy;
 
-    beforeEach(async () => {
+    beforeEach(async(() => {
         TestBed.configureTestingModule({
             imports: [MarqueeWizardModule],
             declarations: [MarqueeWizardValidationComponent]
         }).compileComponents();
-    });
+    }));
 
     beforeEach(() => {
         fixture = TestBed.createComponent(MarqueeWizardValidationComponent);
         component = fixture.componentInstance;
         nativeElement = fixture.nativeElement;
+        visitedChanged = spyOn(fixture.componentInstance, 'visitedChanged');
         fixture.detectChanges();
     });
 
     it('should remove visited state from later steps when valid = false', async () => {
+        // 3 steps and number 2 gets set invalid - 1 should stay the same, 3 should be valid but no longer be visited
         component.step1Visited = true;
         component.step2Visited = true;
+        component.step3Visited = true;
         fixture.detectChanges();
 
         expect(isStepValid(0)).toBe(true);
         expect(isStepValid(1)).toBe(true);
+        expect(isStepValid(2)).toBe(true);
 
         expect(isStepVisited(0)).toBe(true);
         expect(isStepVisited(1)).toBe(true);
+        expect(isStepVisited(2)).toBe(true);
 
-        component.step1Valid = false;
+        component.step2Valid = false;
         fixture.detectChanges();
 
-        expect(isStepValid(0)).toBe(false);
-        expect(isStepValid(1)).toBe(true);
+        expect(isStepValid(0)).toBe(true);
+        expect(isStepValid(1)).toBe(false);
+        expect(isStepValid(2)).toBe(true);
 
         expect(isStepVisited(0)).toBe(true);
-        expect(isStepVisited(1)).toBe(false);
+        expect(isStepVisited(1)).toBe(true);
+        expect(isStepVisited(2)).toBe(false);
+
+        // check visitedChange was emitted 1 time only
+        expect(visitedChanged.calls.all().length).toBe(1);
```

Comment:
Can you assert the values of `setXVisited` in this test please?

Reply:
@donna-marijne Is this in every test or just this one?

Reply:
You can certainly add it to the other test as well if you think it will improve coverage.